clients/go/coreutils/literals.go

```diff
@@ -483,7 +484,17 @@ func MakeLiteralForType(t *core.LiteralType, v interface{}) (*core.Literal, erro
 		}
 	case *core.LiteralType_Simple:
 		newT := t.Type.(*core.LiteralType_Simple)
-		lv, err := MakeLiteralForSimpleType(newT.Simple, fmt.Sprintf("%v", v))
+		strValue := fmt.Sprintf("%v", v)
+		if newT.Simple == core.SimpleType_STRUCT {
+			if _, isValueStringType := v.(string); !isValueStringType {
+				byteValue, err := json.Marshal(v)
+				if err != nil {
+					return nil, fmt.Errorf("unable to marshal to yaml string for struct value %v", v)
```

Comment:
```suggestion
					return nil, fmt.Errorf("unable to marshal to JSON string for struct value %v", v)
```

JSON and YAML have different semantics, so worth being specific. If the input from elsewhere, e.g. flytectl, is YAML originally, then this error should be interpreted there instead. Might be worth returning a specific error or error struct instead here.

Reply:
Thanks for pointing that out. Yes this is just json marshalling issue